files/anitya_cron.py

```diff
@@ -13,11 +12,40 @@
 
 LOG = logging.getLogger('anitya')
 
-def update_project(project):
+
+def indexed_listings():
+    """ Return the full list of project names found by feed listing. """
+    for backend in anitya.lib.plugins.get_plugins():
+        LOG.info("Checking feed for backend %r" % backend)
+        try:
+            for name, homepage, backend, version in backend.check_feed():
+                yield name, homepage, backend, version
+        except NotImplementedError:
+            LOG.debug("Skipping feed check for backend %r" % backend)
+            # Not all backends have the check_feed classmethod implemented,
+            # and that's okay...  Just ignore them for now.
+            continue
+
+
+def projects_by_feed(session):
+    """ Return the list of projects out of sync, found by feed listings.
+
+    If a new entry is noticed and we don't have a project for it, add it.
+    """
+    for name, homepage, backend, version in indexed_listings():
+        project = anitya.lib.model.Project.get_or_create(
+            session, name, homepage, backend)
+        if project.latest_version == version:
```

Comment:
This may check may be false for an already recorded release to a project with multiple stable branches (e.g. a Django 1.8.x security update that's already been detected may be reported again later, since Django 1.9.x is the latest version in absolute terms)